libs/leiden_algorithm/leiden_algorithm/DataBlockingAlgorithm.py

```diff
@@ -13,179 +14,178 @@
+                    + name_table["first_name"].apply(lambda x: get_first_char(x).lower())
+                )
+                short_name_list = name_table["short_name"].drop_duplicates().values
+                name_table = pd.merge(
+                    name_table,
+                    pd.DataFrame(
+                        {
+                            "short_name": short_name_list,
+                            "short_name_id": [
+                                hashlib.sha256(str.encode(x)).hexdigest()
+                                for x in short_name_list
+                            ],
+                        }
+                    ),
+                    on="short_name",
+                    how="left",
+                )
+                return name_table
+    
+            def construct_name_paper_table(author_table, name_table):
+    
+                name_paper_table = author_table.copy().rename(
+                    columns={
+                        "wos_standard": "name",
+                        "UID": "paper_id",
+                        "email_addr": "email_address",
                     }
-                ),
-                on="short_name",
-                how="left",
-            )
-            return name_table
-
-        def construct_name_paper_table(author_table, name_table):
-
-            name_paper_table = author_table.copy().rename(
-                columns={
-                    "wos_standard": "name",
-                    "UID": "paper_id",
-                    "email_addr": "email_address",
-                }
-            )
-            name_paper_table = pd.merge(
-                name_paper_table, name_table[["name_id", "name"]], on="name", how="left"
-            ).drop(columns="name")
-
-            if "_addr_no" is not None:
-                name_paper_table["_addr_no"] = None
-            name_paper_table = slice_columns(
-                name_paper_table,
-                ["name_id", "paper_id", "email_address", "_addr_no", "_seq_no"],
-            )
-
-            def concat(x):
-                s = "_".join(["%s" % v for k, v in x.items()])
-                return str.encode(s)
-
-            name_paper_table["name_paper_id"] = name_paper_table.apply(
-                lambda x: hashlib.sha256(concat(x)).hexdigest(), axis=1
-            )
-            name_paper_table = pd.merge(
-                name_paper_table,
-                name_table[["name_id", "short_name_id", "block_id"]],
-                on="name_id",
-                how="left",
-            )
-            return name_paper_table
-
-        def construct_name_paper_address_table(name_paper_table, address_table):
-
-            # generate name_paper_address_table
+                )
+                
+                name_paper_table = pd.merge(
+                    name_paper_table, name_table[["name_id", "name"]], on="name", how="left"
+                ).drop(columns="name")
+    
+                if "_addr_no" not in name_paper_table:
```

Comment:
Although github did not diff well, this line is also the key in this update. 
Previously, this line was like:

if "_addr_no" is not None:

which is always True and always set all address to be None. Therefore, the rule based on affiliations did not work at all! Gosh